packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -0,0 +1,440 @@
+    MessageValidator.validateSync(jsonLdCredential)
+  }
+
+  public async acceptRequest(
+    agentContext: AgentContext,
+    { credentialFormats, attachId, requestAttachment }: FormatAcceptRequestOptions<JsonLdCredentialFormat>
+  ): Promise<FormatCreateReturn> {
+    const jsonLdFormat = credentialFormats?.jsonld
+
+    // sign credential here. credential to be signed is received as the request attachment
+    // (attachment in the request message from holder to issuer)
+    const credentialRequest = requestAttachment.getDataAsJson<SignCredentialOptionsRFC0593>()
+
+    let verificationMethod = credentialFormats?.jsonld?.verificationMethod
+
+    // FIXME: Need to transform from json to class instance
+    // FIXME: SignCredentialOptions doesn't follow RFC0593
+    // FIXME: Add validation of the jsonLdFormat data (if present)
+    const credentialData = jsonLdFormat ?? credentialRequest
+
+    // FIXME: we're not using all properties from the interface. If we're not using them,
+    // they shouldn't be in the interface.
+    if (!verificationMethod) {
+      verificationMethod = await this.deriveVerificationMethod(
+        agentContext,
+        credentialData.credential,
+        credentialRequest
+      )
+      if (!verificationMethod) {
+        throw new AriesFrameworkError('Missing verification method in credential data')
+      }
+    }
+    const format = new CredentialFormatSpec({
+      attachId,
+      format: JSONLD_VC,
+    })
+
+    const verifiableCredential = await this.w3cCredentialService.signCredential(agentContext, {
+      credential: JsonTransformer.fromJSON(credentialData.credential, W3cCredential),
+      proofType: credentialData.options.proofType,
+      verificationMethod: verificationMethod,
+    })
+
+    const attachment = this.getFormatData(verifiableCredential, format.attachId)
+    return { format, attachment }
+  }
+
+  /**
+   * Derive a verification method using the issuer from the given verifiable credential
+   * @param credential the verifiable credential we want to sign
+   * @return the verification method derived from this credential and its associated issuer did, keys etc.
+   */
+  public async deriveVerificationMethod(
+    agentContext: AgentContext,
+    credential: W3cCredential,
+    credentialRequest: SignCredentialOptionsRFC0593
+  ): Promise<string> {
+    // extract issuer from vc (can be string or Issuer)
+    let issuerDid = credential.issuer
+
+    if (typeof issuerDid !== 'string') {
+      issuerDid = issuerDid.id
+    }
+    // this will throw an error if the issuer did is invalid
+    const issuerDidDocument = await this.didResolver.resolveDidDocument(agentContext, issuerDid)
+
+    // find first key which matches proof type
+    const proofType = credentialRequest.options.proofType
+
+    const keyType: string | undefined = proofTypeKeyTypeMapping[proofType]
+    if (!keyType) {
+      throw new AriesFrameworkError(`No Key Type found for proofType ${proofType}`)
+    }
```

Comment:
The w3cCredentialService now has a method `getKeyTypesByProofType`, it also has a `getVerificationMethodTypesByProofType` which you could use to get the verification method type and pass that to `findVerficationMethodByVerificationMethodType` (modify from `findVerificationMethodByKeyType`)

Reply:
This isn't resovled yet

Reply:
I can't find a method `findVerficationMethodByVerificationMethodType`. Where is that?